applications/sandbox/systems/physx_test_system.cpp

```diff
@@ -0,0 +1,231 @@
+    }
+
+    void PhysXTestSystem::shootPhysXSphere(ShootPhysXSphere& action)
+    {
+        if (!action.value) { return; }
+
+        math::vec3 cameraDirection; math::vec3 cameraPosition;
+        getCameraPositionAndDirection(cameraDirection, cameraPosition);
+
+        auto shiftedBlock = createDefaultMeshEntity(cameraPosition, sphereH, concreteMat);
+
+        {
+            auto shiftedBlockPC = shiftedBlock.add_component<physics_component>();
+            shiftedBlockPC->physicsCompData.AddSphereCollider(0.5f, math::vec3());
+
+            rigidbody& rb = *shiftedBlock.add_component<rigidbody>();
+            rb.rigidbodyData.setVelocity(cameraDirection * 20.0f);
+        }
+
+        self_destruct_component& block = *shiftedBlock.add_component<self_destruct_component>();
+
+        block.selfDestructTimer = 5.0f;
+    }
+
+    void PhysXTestSystem::getCameraPositionAndDirection(math::vec3& cameraDirection, math::vec3& cameraPosition)
+    {
+        //get camera position and set transform to camera postiion 
+        ecs::filter<rendering::camera> cameraQuery;
+
+        //assume the first camera is the player controlled camera
+        for (auto ent : cameraQuery)
+        {
+            auto [positionCamH, rotationCamH, scaleCamH] = ent.get_component<transform>();
+            cameraDirection = rotationCamH.get() * math::vec3(0, 0, 1);
+            cameraPosition = positionCamH.get() + cameraDirection;
+        }
+    }
+
+    void PhysXTestSystem::initializeLitMaterial(rendering::material_handle& materialToInitialize,
+        rendering::shader_handle& litShader, const fs::view& albedoFile, const fs::view& normalFile, const fs::view& roughnessFile)
+    {
+        static size_t litCount = 0;
+
+        materialToInitialize = rendering::MaterialCache::create_material("LitMaterial" + litCount, litShader);
+        materialToInitialize.set_param("alphaCutoff", 0.5f);
+        materialToInitialize.set_param("useAlbedoTex", true);
+        materialToInitialize.set_param("useRoughnessTex", true);
+        materialToInitialize.set_param("useNormal", true);
+
+        materialToInitialize.set_param("useEmissiveTex", false);
+        materialToInitialize.set_param("useAmbientOcclusion", false);
+        materialToInitialize.set_param("useHeight", false);
+        materialToInitialize.set_param("useMetallicTex", false);
+        materialToInitialize.set_param("useMetallicRoughness", false);
+
+        materialToInitialize.set_param("metallicValue", 0.0f);
+        materialToInitialize.set_param("emissiveColor", math::colors::black);
+
+        materialToInitialize.set_param("albedoTex", rendering::TextureCache::create_texture(albedoFile));
+        materialToInitialize.set_param("normalTex", rendering::TextureCache::create_texture(normalFile));
+        materialToInitialize.set_param("roughnessTex", rendering::TextureCache::create_texture(roughnessFile));
+        materialToInitialize.set_param("skycolor", math::color(0.1f, 0.3f, 1.0f));
```

Comment:
latest PR that got merged makes this line unnecessary, we don't have sky colors anymore (yay!)